Issue 1042 and 1037

juzen2003 opened the pull request:
- Fixes #1037, #1042 
- Were any Django, import pipeline, table_schema, or dictionary files modified? N
  - Database used: opus3_test_200517
  - All Django tests pass: Y
- Were any JavaScript or CSS files modified? Y
  - JSHINT run on all affected files: Y (widgets.js)
  - Tested on Chrome, Firefox, Safari, and iOS: Y
    (Remember to test all browser sizes)
  - Tested for race conditions (with delayed API calls if applicable): Y

Description of changes:
- Fixed #1037 by encoding the input value before passing into stringsearchchoices API URL.
- Fixed #1042 by making sure when RANGE widgets (without qtype & unit inputs) just get open, the corresponding slugs are updated in opus.selections properly. Previously they are not in opus.selections. 

Known problems:
None

juzen2003 committed:
Fix issue 1042 by properly updating opus.selections when a RANGE widget
without qtype & unit just got open.
- In widgets.js (line 1368-1379), when a RANGE widget that has no qtype
  & unit inputs just got open, update opus.selections to make sure input
  slugs exist in opus.selections.

juzen2003 committed:
Fix issue 1042 by encoding the input value passed into stringsearchchoices
API url.
- In widgets.js (line 1597-1599), encode the input value before passing
  into stringsearchchoices API url.
- In widgets.js (line 1585-1587), clean up the redundant "/".

rfrenchseti commented on opus/application/static_media/js/widgets.js:
```diff
@@ -1365,6 +1365,18 @@ var o_widgets = {
                 o_search.getHinting(slug);
             }
+
+            // if a RANGE widget that has no qtype & unit inputs just got open, we need to
```
Why is this limited to widgets with no qtype and units? Even if the other cases are taken care of in alignDataInSelectionsAndExtras it's very confusing to have this limitation here with no explanation. Wouldn't it be simpler to just always do it?

juzen2003 replied:
Just updated it to always check & update opus.selections when input widgets just got opened. Previously I limited to widgets with no qtype & units because opus.selections of other cases will be updated in alignDataInSelectionsAndExtras. But you are right, it's confusing and looks like a special case by only doing this step for widgets with no qtype & units there. Now we do the step (update opus.selections) for all cases of input widgets.

rfrenchseti commented on opus/application/static_media/js/widgets.js:
```diff
@@ -1365,6 +1365,18 @@ var o_widgets = {
                 o_search.getHinting(slug);
             }
+
+            // if a RANGE widget that has no qtype & unit inputs just got open, we need to
```
open => opened

juzen2003 committed:
Update opus.selections to make sure input slugs exist when an input
widget just got opened.
- In widgets.js (line 1369-1381), make sure input slugs exist in
  opus.selections when an input widget just got opened.

juzen2003 committed:
Fix the comment at line 1369 of widgets.js.
